source/common/matcher/matcher.h

```diff
@@ -65,6 +65,17 @@ using MatchTreeFactoryCb = std::function<std::unique_ptr<MatchTree<DataType>>()>
 template <class DataType> using OnMatchFactoryCb = std::function<OnMatch<DataType>()>;
 template <class DataType> using DataInputFactoryCb = std::function<DataInputPtr<DataType>()>;
 
+template <class DataType> class AnyMatcher : public MatchTree<DataType> {
+public:
+  explicit AnyMatcher(absl::optional<OnMatch<DataType>> on_no_match)
+      : on_no_match_(std::move(on_no_match)) {}
+
+  typename MatchTree<DataType>::MatchResult match(const DataType&) override {
+    return {MatchState::MatchComplete, on_no_match_};
```

Comment:
I'm not an expert here, but should the 2nd param (`on_no_match_` object) be returned by ref?

Reply:
I think in other use cases this always does a copy, since the Action object is generally small (lambda + pointer). We could probably change this around but would be a bigger refactor I think